wrapper/src/main/java/software/amazon/jdbc/wrapper/ConnectionWrapper.java

```diff
@@ -139,15 +146,34 @@ protected void init(
     }
   }
 
-  protected String getProtocol(final String url) {
+  protected DatabaseDialect getDialect(final String url, Properties info) {
+    String dialectClassName = DATABASE_DIALECT.getString(info);
+    if (dialectClassName != null) {
+      try {
+        return (DatabaseDialect) Class.forName(dialectClassName).getDeclaredConstructor().newInstance();
+      } catch (Exception ex) {
+        // ignore if we don't have a dialect in the properties we will try to figure it out from the
+        // URL scheme
+      }
+    }
     final int index = url.indexOf("//");
     if (index < 0) {
       throw new IllegalArgumentException(
           Messages.get(
               "ConnectionWrapper.protocolNotFound",
               new Object[] {url}));
     }
-    return url.substring(0, index + 2);
+    switch (url.substring(0, index + 2)) {
```

Comment:
I'd suggest to create a Dialect Manger class and hide this logic there.

Reply:
Ya, that was pretty ugly :)

Reply:
It seems that protocol is not enough to determine what dialect class to create. As I noted above. Aurora dialect and vanilla MySQL dialects are different but they share the same protocol.